### PR TITLE
Block header support

### PR DESCRIPTION
**Pull request by afk11**
Makes use of #32 

Adds basic support for block headers (serialization and hashing)

### CODE REVIEW

File: src/block.c [hunk 1]
```diff
@@ -0,0 +1,113 @@
+/*
+
+ The MIT License (MIT)
+
+ Copyright (c) 2015 Jonas Schnelli
```
Comment (jonasschnelli) [inline src/block.c]: Use your name?

**Comment by jonasschnelli** [comment]
Nice! Thanks.
Concept ACK. Will test soon.

File: src/block.c (hunk 1 above)
Comment (afk11) [inline src/block.c]: Sure will push later, just copy pasted!

**Comment by jonasschnelli** [comment]
This is saccutally a nice PR! Thanks again.

Once the `ser_i32` stuff is fixed, I think then this PR is ready to merge.

**Comment by jonasschnelli** [comment]
utACK

**Pull request merged**